shell/edgexfoundry-snapcraft.sh

```diff
@@ -0,0 +1,175 @@
+#!/bin/bash -x
+
+# note that since this script gets included into a yaml string using jjb's
+# include-raw-escape macro, we can't use any squiggly braces here, otherwise
+# they will get duplicated
+# for some reason we also can't use include-raw macro because that macro will
+# attempt to interpret anything contianed in a macro as a jenkins key to
+# expand which we also don't want
+
+# if we are running inside a jenkins instance then copy the login file 
+# and check if this is a release job
+if [ -n "$JENKINS_URL" ]; then
+    # since we are running from jenkins, use the workspace as the git root
+    GIT_ROOT=$WORKSPACE
+    if [ -f "$HOME/EdgeX" ]; then
+       cp "$HOME/EdgeX" "$GIT_ROOT/edgex-snap-store-login"
+    else
+        echo "I seem to be running on Jenkins, but there's not a snap store login file..." 
+    fi
+
+    # figure out what kind of job this is using $JOB_NAME and simplify that 
+    # into $JOB_TYPE
+    if [[ "$JOB_NAME" =~ .*-snap-.*-stage-snap.* ]]; then
```

Comment:
Does this and the regex two lines below need to include "snap"? If the script is running, can we not assume we are in the correct job type?

Reply:
Yes the regex could leave out snap and still be fine, I assume you mean something like

```bash
    if [[ "$JOB_NAME" =~ .*-stage-snap.* ]]; then
        JOB_TYPE="stage"
    elif [[ "$JOB_NAME" =~ .*-release-snap ]]; then
        JOB_TYPE="release"
    else
        JOB_TYPE="build"
    fi
```

?

Reply:
I meant if could be just `.*-stage-.*` and `.*-release-.*`

Reply:
Ah, yes it could be. Do you want me to change it to use the simpler regex?

Reply:
To me, simpler means the code is more generalized and adaptable. My preference would be to make it as generic as possible, so long as it won't create false positives.

Reply:
Hmm, well I guess after thinking about it some more I would rather leave it at `*-snap-.*-stage-snap.*` and `.*-release-snap` because while yes the script won't ever do anything other than run snap jobs, we may at some point have a repo name like `device-stage` where the job names would then look like `device-stage-snap-release-snap` and `device-stage-snap-master-stage-snap` in which case the simpler regex would cause false positives. I agree with the sentiment of having simpler and generic as possible, but I think stricter is better in this instance.

Reply:
Ok, that's acceptable to me.